File: server/api/routes/authRoutes.js
const express = require('express');
const {
  authUser,
  getUserProfile,
  updateUserProfile,
  getAdminEmployees,
  getEmployeeById,
  createAdminEmployee,
  updateAdminEmployee,
  deleteAdminEmployee,
} = require('../controllers/authController.js');
const { protect, admin } = require('../middleware/authMiddleware.js');

const router = express.Router();

// Only admin login is available (users login through admin login endpoint)
router.post('/login', authUser);
router.get('/profile', protect, getUserProfile);
router.put('/profile', protect, updateUserProfile);

// Employee management routes (admin only)
router.get('/employees', protect, admin, getAdminEmployees);
router.get('/employees/:id', protect, admin, getEmployeeById);
router.post('/employees', protect, admin, createAdminEmployee);
router.put('/employees/:id', protect, admin, updateAdminEmployee);
router.delete('/employees/:id', protect, admin, deleteAdminEmployee);

module.exports = router;

File: server/api/routes/auth_routes.py
from fastapi import APIRouter, Depends, HTTPException
from controllers.auth_controller import (
    auth_user,
    get_user_profile,
    update_user_profile,
    get_admin_employees,
    get_employee_by_id,
    create_admin_employee,
    update_admin_employee,
    delete_admin_employee,
)
from middleware.auth_middleware import protect, admin

router = APIRouter()

# Only admin login is available (users login through admin login endpoint)
@router.post("/login")
async def login(data: dict):
    return await auth_user(data)

@router.get("/profile")
async def profile(user=Depends(protect)):
    return await get_user_profile(user)

@router.put("/profile")
async def update_profile(data: dict, user=Depends(protect)):
    return await update_user_profile(data, user)

# Employee management routes (admin only)
@router.get("/employees")
async def employees(user=Depends(admin)):
    return await get_admin_employees(user)

@router.get("/employees/{id}")
async def employee_by_id(id: str, user=Depends(admin)):
    return await get_employee_by_id(id, user)

@router.post("/employees")
async def create_employee(data: dict, user=Depends(admin)):
    return await create_admin_employee(data, user)

@router.put("/employees/{id}")
async def update_employee(id: str, data: dict, user=Depends(admin)):
    return await update_admin_employee(id, data, user)

@router.delete("/employees/{id}")
async def delete_employee(id: str, user=Depends(admin)):
    return await delete_admin_employee(id, user)